File: app/java/MoviesApp/app/src/test/java/com/example/moviesapp/data/movies/impl/MoviesDtoConverterTest.java
package com.example.moviesapp.data.movies.impl;

import com.example.moviesapp.data.movies.dto.MovieDto;
import com.example.moviesapp.data.movies.dto.MoviesListDto;
import com.example.moviesapp.data.movies.model.Movie;

import org.junit.Before;
import org.junit.Test;

import java.util.Collections;
import java.util.List;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNotNull;

public class MoviesDtoConverterTest {

    private MoviesDtoConverter classUnderTest;

    @Before
    public void setup() {
        classUnderTest = new MoviesDtoConverter();
    }

    @Test
    public void apply() {
        String id= "1";
        String title = "title";
        String imageUrl = "imageUrl";
        MovieDto movieDto = new MovieDto(
                id,
                title,
                imageUrl
        );
        MoviesListDto moviesListDto = new MoviesListDto(Collections.singletonList(movieDto));

        List<Movie> result = classUnderTest.apply(moviesListDto);

        assertNotNull(result);
        assertEquals(
                1,
                result.size()
        );

        Movie movie = result.get(0);

        assertNotNull(movie);
        assertEquals(
                id,
                movie.getId()
        );
        assertEquals(
                title,
                movie.getTitle()
        );
        assertEquals(
                imageUrl,
                movie.getImageUrl()
        );
    }
}

File: app/android/kotlin/MoviesApp/app/src/test/java/com/example/moviesapp/data/movies/impl/MoviesDtoConverterTest.kt
package com.example.moviesapp.data.movies.impl

import com.example.moviesapp.data.movies.dto.MovieDto
import com.example.moviesapp.data.movies.dto.MovieStatisticsDto
import com.example.moviesapp.data.movies.dto.MoviesListDto
import com.example.moviesapp.data.movies.model.Movie
import org.junit.Assert.assertEquals
import org.junit.Assert.assertNotNull
import org.junit.Before
import org.junit.Test

class MoviesDtoConverterTest {

    private lateinit var classUnderTest: MoviesDtoConverter

    @Before
    fun setup() {
        classUnderTest = MoviesDtoConverter()
    }

    @Test
    fun invoke() {
        val id = "1"
        val title = "title"
        val imageUrl = "imageUrl"
        val score = 95.0
        val percentViewed = 98.0
        val movieDto = MovieDto(
                id,
                title,
                imageUrl,
                MovieStatisticsDto(
                        score,
                        percentViewed
                )
        )
        val moviesListDto = MoviesListDto(listOf(movieDto))

        val result = classUnderTest(moviesListDto)

        assertNotNull(result)
        assertEquals(
                1,
                result.size
                        .toLong())

        val movie: Movie = result[0]

        with(movie) {
            assertNotNull(this)
            assertEquals(
                    id,
                    this.id
            )
            assertEquals(
                    title,
                    this.title
            )
            assertEquals(
                    imageUrl,
                    this.imageUrl
            )
            with(statistics) {
                assertNotNull(this)
                assertEquals(
                        score,
                        this.score
                )
                assertEquals(
                        percentViewed,
                        this.percentViewed
                )
            }
        }
    }
}